app/apollo/resolvers/subscription.js

```diff
@@ -462,34 +489,56 @@ const subscriptionResolvers = {
         const kubeOwnerId = await models.User.getKubeOwnerId(context);
 
         const conditions = await getGroupConditionsIncludingEmpty(me, org_id, ACTIONS.READ, 'name', queryName, context);
-        logger.debug({req_id, user, org_id, conditions }, `${queryName} group conditions are...`);
+        logger.debug({req_id, user, org_id, conditions}, `${queryName} group conditions are...`);
 
         const subscription = await models.Subscription.findOne({ org_id, uuid, ...conditions }, {}).lean({ virtuals: true });
-        if(!subscription){
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - found: ${!!subscription}`);
+
+        const identifiers = subscription ? [uuid, subscription.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.SUBSCRIPTION, queryName, context, identifiers);
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);
+
+        // find the channel
+        const channel = await models.Channel.findOne({ org_id, uuid: channel_uuid });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${!!channel}`);
+
+        const channelIdentifiers = channel ? [channel_uuid, channel.name] : [channel_uuid];
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - channel authorized`);
+
+        // find the groups
+        const requestedGroups = await models.Group.find({
+          org_id,
+          $or: [
+            { name: { $in: groups } },
+            { uuid: { $in: groups } }
+          ]
+        }).lean({ virtuals: true });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${requestedGroups.length}`);
+
+        const allowedGroups = await filterResourcesToAllowed(me, org_id, ACTIONS.READ, TYPES.GROUP, requestedGroups, context);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - allowed: ${allowedGroups.length}`);
+
+        const groupNames = _.map(allowedGroups, group => group.name);
+
+        // Check for errors after validation
+        if( allowedGroups.length < requestedGroups.length ) {
+          throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found'));
+        }
+        if (!subscription) {
```

Comment:
See earlier comment in `addSubscription` about `allowedGroups` -- same applies here.

Reply:
I changed requestedGroups to foundGroups and am adding this to accommodate:
        if(foundGroups.length < groups.length || allowedGroups.length < groups.length) { throw new NotFoundError(context.req.t('One or more of the passed group uuids were not found')); }

Reply:
resolved, but see new comment below.